lib/avalon/batch/entry.rb

```diff
@@ -176,17 +176,23 @@ def self.offset_valid?( offset )
         true
       end
 
-      def self.attach_datastreams_to_master_file( master_file, filename )
+      def self.attach_datastreams_to_master_file( master_file, filename, captions )
         structural_file = "#{filename}.structure.xml"
         if FileLocator.new(structural_file).exist?
           master_file.structuralMetadata.content=FileLocator.new(structural_file).reader
           master_file.structuralMetadata.original_name = structural_file
         end
-        captions_file = "#{filename}.vtt"
-        if FileLocator.new(captions_file).exist?
-          master_file.captions.content=FileLocator.new(captions_file).reader
-          master_file.captions.mime_type='text/vtt'
-          master_file.captions.original_name = captions_file
+        captions.each do |c|
+          return unless c.present?
```

Comment:
This appears to skip any subsequent captions if there is a nil in the list.  What do you think about skipping instead?
```suggestion
          next unless c.present?
```

Reply:
You could also probably combine this line with the next.
```suggestion
          next unless c.present? && c[:caption_file].present? && FileLocator.new(c[:caption_file]).exist?
```

Reply:
Yeah, that should definitely be a `next`. In the moment I forgot that early return breaks out of loops. 

And moving the following `if` statement into the guard clause makes sense :+1: